src/main/java/ch/njol/skript/effects/EffBroadcast.java

```diff
@@ -57,30 +63,37 @@
 
 	@SuppressWarnings("unchecked")
 	@Override
-	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parser) {
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		messages = LiteralUtils.defendExpression(exprs[0]);
 		worlds = (Expression<World>) exprs[1];
 		return LiteralUtils.canInitSafely(messages);
 	}
 	
 	@Override
+	@SuppressWarnings("deprecation")
 	public void execute(Event e) {
-		World[] worlds = this.worlds != null ? this.worlds.getArray(e) : null;
-
-		for (Object object : messages.getArray(e)) {
-			String string = object instanceof String ? (String) object : Classes.toString(object);
+		List<CommandSender> receivers = new ArrayList<>();
+		if (worlds == null) {
+			receivers.addAll(Bukkit.getOnlinePlayers());
+			receivers.add(Bukkit.getConsoleSender());
+		} else {
+			for (World world : worlds.getArray(e))
+				receivers.addAll(world.getPlayers());
+		}
 
-			if (worlds == null) {
-				// not Bukkit.broadcastMessage to ignore permissions
-				for (Player player : PlayerUtils.getOnlinePlayers()) {
-					player.sendMessage(string);
-				}
-				Bukkit.getConsoleSender().sendMessage(string);
-			} else {
-				for (World w : worlds) {
-					for (Player player : w.getPlayers()) {
-						player.sendMessage(string);
-					}
+		if (messages instanceof ExprColoured && ((ExprColoured) messages).isUnsafeFormat()) { // Manually marked as trusted
+			for (Object message : messages.getArray(e)) {
+				BaseComponent[] components = BungeeConverter.convert(ChatMessages.parse((String) message));
+				receivers.forEach(receiver -> receiver.spigot().sendMessage(components));
+			}
+		} else {
+			for (Object message : messages.getArray(e)) {
+				if (message instanceof VariableString) {
```

Comment:
VariableString is an expression, and expression's getter (e.g. `message.getArray(e)`) will not return another expression